dashboard/app/models/levels/blockly.rb

```diff
@@ -504,25 +504,57 @@ def localized_function_blocks(blocks)
         default: nil,
         smart: true
       )
-      function_description = function.at_xpath('./mutation/description')
+      original_function_name = function_name.content
+      function_name.content = localized_name if localized_name
+      function_mutation = function.at_xpath('./mutation')
+      next unless function_mutation
+      function_description = function_mutation.at_xpath('./description')
       localized_description = I18n.t(
         "description",
-        scope: [:data, :function_definitions, name, function_name.content],
+        scope: [:data, :function_definitions, name, original_function_name],
         default: nil,
         smart: true
       )
-      function_name.content = localized_name if localized_name
       function_description.content = localized_description if localized_description
+      function_mutation.xpath("./arg").each do |parameter|
+        localized_parameter = I18n.t(
+          parameter["name"],
+          scope: [:data, :function_definitions, name, original_function_name, "parameters"],
+          default: nil,
+          smart: true
+        )
+        parameter["name"] = localized_parameter if localized_parameter
+      end
+      function.xpath(".//title[@name=\"VAR\"]").each do |parameter|
```

Comment:
Could you leave some inline comments explaining why it seems there are two blocks of code which translate parameter names? It looks like the reason is because parameters are defined in two different xpaths, but I'm not sure.